### PR TITLE
Fix multiple issues on MacOS + keypress events on windows not UTF

### PR DESCRIPTION
**Pull request by tyronx**
### Purpose of this PR

As requested on discord and on twitter (https://twitter.com/Dylan_Perks/status/1132017054515650560) I cherry picked my commits into a freshly forked 3.0 branch.

As mentioned multiple times on discord and various PRs the mac os input system is broken in various ways, some other bugs, as well as windows keypress events not being utf8 encoded. See commits for explanations. 

### Testing status

* OpenTK 3.1/Anego fork, which is compiled with these commits, is running on around 1000 users as part of my game Vintage Story, so seems stable enough at least for my use cases.
* The MacOS fixes make my game playable on a macbook air and a macbook pro. I've heard reports there is more macos input issues that I have yet to track down, but at least the game is now playable for some of my Mac user base

I'm happy to send out a couple of game keys to OpenTK devs to verify the results, or to just try out Vintage Story, if that helps with anything. PM me on discord.

### Comments

Here should be a full description of all changes, but I don't want spend yet another 2 hours to pick out all the individual issues that were discussed with the uncertain outcome of this PR, judging by the past history of my own and related PRs. 

I do would like to take this opportunity to say that, while I really appreciate everyone's continued efforts in building a great open source product at the same time I also feel highly disgruntled by the fact that the leadership decided to quasi abandon thousands upon thousands of users by not accepting important bugfixes for the **only stable** release build of OpenTK. And I mean that literally, because the pc game I am working on is unplayable with OpenTK 3.0 on MacOS.

And now, seemingly on a whim its worth supporting after all? Perhaps I didn't follow the discussions on discord closely enough but I cannot see any lengthy discussion or explanation as to why it's suddenly worth doing a maintenance update now, but when I reported those problems in Fall 2018 it was not?

I'd like to contribute to the project with bugfixes, but leaving stuff broken and having PRs denied is highly demotivating, hence if this PR is denied again, I'm unlikely to do any continued efforts towards this and just keep working on my own fork of OpenTK.

Sorry if this comes across as harsh, but I really just want to build a great game and not meddle with 3rd party library politics :/

### CODE REVIEW

**Comment by varon** [comment]
@tyronx - It seems like there's been an epic miscommunication that happened some time ago. Despite some confusion around the specifics, I've always intended for us to encourage and accept community contributions.

We have chosen to focus our own development efforts onto 4.0, but **we are absolutely willing to accept fixes and non-feature improvements for 3.x**. OpenTK 3.x is a widely used library - its community is central to it's success. We're not perfect, but we aim to provide the best stewardship we can, so feedback like this is extremely valuable.

Thanks for this excellent and thorough PR. This is a great candidate for inclusion in the 3.0 maintenance release.

**Comment by tyronx** [comment]
Happy to hear that you still accept fixes. For reference, this is what you yourself wrote on discord 2 months ago - not sure what part of "3.0 is frozen" can be considered miscommunication there. 

```
Perksey 03/28/2019
The current stance that we've been working under is that the 3.0 branch is frozen.

Varon 03/28/2019
Yes, it is frozen

[....]

Perksey 03/28/2019
oh that one
Alright. I guess if he championed it we could release a final 3.0 version. The assumption I've been operating under is that we were no longer accepting PRs as we've frozen the branch in an effort to draw contributors to the 4.0 branch. We've rejected past PRs because of this, so I'm a little confused as to when this stance changed?
I don't wanna send mixed messages to the community.
```

**Comment by varon** [comment]
As I said a lot of confusion here. The original conversation you're quoting actually included me offering to push out a patch release if a PR was made for the issue in question.

```
[00:42] Perksey: but the question is will we release a patch for it?
[00:42] Varon: I'm happy to do that
```

Frozen was intended to mean that the active OpenTK developers aren't accepting new features for 3.0, not that we'd reject any fixes. Probably wasn't the best choice of words.  An example of this is that we declined to backport the integer vectors from 4.0 to 3.0.

Hope that clarifies things.

**Comment by leezer3** [comment]
Massive thanks from me.

Some of it is fixing my cruddy code, but I have absolutely no desire to leave 3.0 at the minute.

File: src/OpenTK/Platform/Linux/LinuxNativeWindow.cs [hunk 1]
```diff
@@ -500,22 +500,38 @@ public override Size ClientSize
             }
         }
 
-        public override bool CursorVisible
+        public override bool CursorGrabbed
```
Comment (varon) [inline src/OpenTK/Platform/Linux/LinuxNativeWindow.cs]: Can you elaborate a bit on this change?

File: src/OpenTK/Platform/NativeWindowBase.cs [hunk 2]
```diff
@@ -188,8 +189,8 @@ protected void UpdateModifierFlags(KeyModifiers mods)
 
             if (control)
             {
-                OnKeyDown(Key.ControlLeft, KeyboardState[Key.AltLeft]);
-                OnKeyDown(Key.ControlRight, KeyboardState[Key.AltLeft]);
+                OnKeyDown(Key.ControlLeft, KeyboardState[Key.ControlLeft]);
+                OnKeyDown(Key.ControlRight, KeyboardState[Key.ControlRight]);
```
Comment (varon) [inline src/OpenTK/Platform/NativeWindowBase.cs]: Nice work finding these bugs.

File: src/OpenTK/Platform/Linux/LinuxNativeWindow.cs (hunk 1 above)
Comment (jvbsl) [reply src/OpenTK/Platform/Linux/LinuxNativeWindow.cs]: don't let yourself be mislead by git diff, It seems to me that CursorVisible is still the same, but CursorGrabbed got added.

Comment (tyronx) [reply src/OpenTK/Platform/Linux/LinuxNativeWindow.cs]: These changes are part of PR #589 which I cherry picked into my fork. If I remember correctly it was necessary to fix some issues an mac os as well as on linux, but @copygirl should know more about the linux part.

**Comment by tyronx** [comment]
Anything you'd like me to do on my end still? Shall I add some change log stuff to the RELEASE_NOTES.md? If yes, what version number is planned for the next 3.x release?

**Comment by Perksey** [comment]
@tyronx Yes please, the last 3.X release will be 3.1.

**Pull request merged**